tests/reexecute/c/vm_reexecute_test.go

```diff
@@ -104,7 +104,16 @@ func TestMain(m *testing.M) {
 	flag.IntVar(&chanSizeArg, "chan-size", 100, "Size of the channel to use for block processing.")
 	flag.DurationVar(&executionTimeout, "execution-timeout", 0, "Benchmark execution timeout. After this timeout has elapsed, terminate the benchmark without error. If 0, no timeout is applied.")
 
-	flag.BoolVar(&metricsServerEnabledArg, "metrics-server-enabled", false, "Whether to enable the metrics server.")
+	flag.Func("metrics-server-port", "Starts a metrics server and sets the port it will listen to", func(s string) error {
+		port, err := strconv.ParseUint(s, 10, 64)
+		if err != nil {
+			return err
+		}
+
+		metricsServerPort = new(uint64)
+		*metricsServerPort = port
+		return nil
+	})
```

Comment:
Isn't there a way to simply check if a flag is set without using a Func like this?

Reply:
Ah I was thinking of the viper style syntax which supports differentiating if a flag was explicitly set or not.